### ISSUE TITLE
batch norm in conditional output branch?

### ISSUE THREAD

**Issue by gdshen** [jalola/improved-wgan-pytorch#6]
hi, i know in the origin wgan-gp implementation we can not use batch norm, i just wonder whether we can use batch norm in the conditional output branch?

**Comment by jalola** [comment]
Hi,

In the Residual block, we used BatchNorm (basically, Generator uses it but Discriminator uses Layer Norm). 
https://github.com/jalola/improved-wgan-pytorch/blob/master/models/wgan.py#L94

In the Tensorflow implementation, the author also used BatchNorm:
https://github.com/igul222/improved_wgan_training/blob/master/gan_64x64.py#L189

**Comment by gdshen** [comment]
Thanks for the links. 

And i am confused about
https://github.com/jalola/improved-wgan-pytorch/blob/5bf75f8924a65ac141e360debee8ef15b157abd4/models/conwgan.py#L222
whether can we add batch norm after "output_gan",  since this branch has no relation with the wasserstein distance.

For example

```python
# in the self.__init__
self.bn = nn.BatchNorm1d(feature_nums)

...
# in the self.forward
output_congan = self.ln2(output)
output_congan = self.bn(output_congan)
```

**Comment by jalola** [comment]
Sorry for misunderstanding your question. I got it now.

However, I am not experienced in using BatchNorm and I rarely see people use BatchNorm at the end of the output. There, we want the output to be a label (0, 1, 2 ...) of image input so we can compute the loss with real label (https://github.com/jalola/improved-wgan-pytorch/blob/master/congan_train.py#L253).
So I think we don't need to use BatchNorm.
You can try it out and let us know about that.
